Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -584,15 +583,21 @@
 
     m_page->chrome().client().removeInitialTextAnimationForActiveWritingToolsSession();
 
-    auto addDestinationTextAnimation = [weakThis = WeakPtr { *this }, attributedText, range, context, finished](TextAnimationRunMode runMode) mutable {
-        weakThis->compositionSessionDidReceiveTextWithReplacementRangeAsync(attributedText, range, context, finished, runMode);
+    // Must generate these UUID now to pass into the source animation for iOS to work.
+    auto sourceAnimationUUID = WTF::UUID::createVersion4();
+    auto destinationAnimationUUID = WTF::UUID::createVersion4();
+
+    auto addDestinationTextAnimation = [weakThis = WeakPtr { *this }, attributedText, range, context, finished, sourceAnimationUUID, destinationAnimationUUID](TextAnimationRunMode runMode) mutable {
+        weakThis->compositionSessionDidReceiveTextWithReplacementRangeAsync(sourceAnimationUUID, destinationAnimationUUID, attributedText, range, context, finished, runMode);
     };
 
-#if PLATFORM(MAC)
-    m_page->chrome().client().addSourceTextAnimationForActiveWritingToolsSession(range, attributedText.string, WTFMove(addDestinationTextAnimation));
-#else
-    addDestinationTextAnimation(WebCore::TextAnimationRunMode::RunAnimation);
-#endif
+    // We only get a single replace call for smart replies with finished = true. We use this flag to not run the final replace for a compsition
+    // session, so for smart replies, we need to make sure to not send with this flag, so that we can be sure to do the animation for smart replies.
+    if (session.compositionType == WritingTools::SessionCompositionType::SmartReply)
+        finished = false;
+
+    m_page->chrome().client().addSourceTextAnimationForActiveWritingToolsSession(sourceAnimationUUID, destinationAnimationUUID, finished, range, attributedText.string, WTFMove(addDestinationTextAnimation));
```

Comment:
can't the uuids be made within `addSourceTextAnimationForActiveWritingToolsSession`, and then just call `addDestinationTextAnimation` with those as the parameters?

Reply:
No, we need them to be passed in to each function call, and to be created outside of completion handlers. I tried to do what you describe, but it didn't work.